Neo4jClient.DataAnnotations/CypherFluentQueryExtensions.cs

```diff
@@ -416,7 +416,7 @@ public static ICypherFluentQuery Set(this ICypherFluentQuery query, string varia
         /// <returns></returns>
         public static ICypherFluentQuery Set(this ICypherFluentQuery query, string variable, Expression<Func<object>> properties, out string setParameter)
         {
-            return SharedSet(query, variable, properties, PropertiesBuildStrategy.WithParams, out setParameter, add: false);
+            return SharedSet(query, variable, properties, PropertiesBuildStrategy.WithParamsForValues, out setParameter, add: false);
```

Comment:
Same comment as with "Create" here too.

Reply:
I changed it when i was tested the set and create while attempting to add an entity to the database.
Apparently without WithParamsForValues, creation in neo4j won't work with the latest version, as they require the object to be serialized as a map (key, value) when creating a parameter.
While looking for a work around, i found that you supported this using that option.
After testing it extensively it seemed to work fine.

Let me know if there is anything else i can help with.

Reply:
Thanks for the proposition, i would definitely like that, I understand that work sometimes can be demanding and would consume the majority of our time. I would like to contribute to the project with you on the various points you mentioned and other features ahead in both projects.

Thanks you for your trust.

Reply:
Cool. I will add you as soon as I'm done with this release from your PR.

However, I just checked the documentation for CREATE and SET on Neo4j docs and it seems parameters may still be supported directly on those statements (see [here](https://neo4j.com/docs/cypher-manual/4.1/clauses/create/#use-parameters-with-create) and [here](https://neo4j.com/docs/cypher-manual/4.1/clauses/set/#set-set-all-properties-using-a-parameter)). 

So can you please share the exact error you were having here, I want to make sure that it isn't a bug in the library that's causing the error in the first place. Thanks.

Reply:
I've recorded a loom video to show you what happens exactly when i use WithParams.

https://www.loom.com/share/a9157aed8f0242e8b8bc5dd53bf14828

Let me know if there is anything else i can do, or if i fixed it the wrong way.

Thanks

Reply:
Thanks for the video, it explained quite some. 

This error is most likely from the Neo4jClient library, as I can see the error says it expected map but got List<T> instead. Creating parameters from the annotations library was always a little hack due to access level protection of Neo4jClient. 

I will do a dive into the Neo4jClient code to know what changed in the latest package and fix this.

Reply:
Great, thanks, let me know if there is anything i can do.

Reply:
Hi @Arsslensoft 

So the issue here was that the underlying Neo4j Driver converts JObject to a List<Token>, instead of Dictionary. That resulted in issues when the object is serialized because Neo4j was getting a list instead of a map. 

I initially noticed this when I added Bolt support and tried to fix it. Seems I missed a few spots. The issue itself is only now showing up because Neo4jClient uses the Neo4j Driver directly almost everywhere it could for its Bolt client, and tries not to interface.

All fixed now, and a new package, 2.2.1, released.

Thanks for the support provided.